server/controller/src/main/java/ai/starwhale/mlops/api/EvaluationController.java

```diff
@@ -127,4 +141,96 @@ ResponseEntity<ResponseMessage<PageInfo<SummaryVo>>> listEvaluationSummary(
+            method = RequestMethod.HEAD,
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @PreAuthorize("hasAnyRole('OWNER', 'MAINTAINER', 'GUEST')")
+    ResponseEntity<?> headHashedBlob(
+            @PathVariable(name = "projectUrl") String projectUrl,
+            @PathVariable(name = "version") String version,
+            @PathVariable(name = "hash") String hash
+    ) {
+        HashNamedObjectStore hashNamedObjectStore = evaluationFileStorage.hashObjectStore(projectUrl, version);
+        String path;
+        try {
+            path = hashNamedObjectStore.head(hash);
+        } catch (IOException e) {
+            log.error("access to main object storage failed", e);
+            throw new SwProcessException(ErrorType.STORAGE, "access to main object storage failed", e);
+        }
+        if (null != path) {
+            return ResponseEntity.ok().header("X-SW-LOCAL-STORAGE-URI", path).build();
+        } else {
+            return ResponseEntity.notFound().build();
+        }
+    }
+
+    @Operation(summary = "Download the hashed blob in this evaluation",
+            description = "404 if not exists; 200 if exists")
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "ok")})
+    @RequestMapping(
+            value = "/project/{projectUrl}/evaluation/{version}/hashedBlob/{hash}",
+            method = RequestMethod.GET,
+            produces = MediaType.APPLICATION_OCTET_STREAM_VALUE)
+    @PreAuthorize("hasAnyRole('OWNER', 'MAINTAINER', 'GUEST')")
+    void getHashedBlob(
+            @PathVariable(name = "projectUrl") String projectUrl,
+            @PathVariable(name = "version") String version,
+            @PathVariable(name = "hash") String hash,
+            HttpServletResponse httpResponse
+    ) {
+        try (
+                var inputStream = evaluationFileStorage.hashObjectStore(projectUrl, version).get(hash.trim());
+                var outputStream = httpResponse.getOutputStream()
+        ) {
+            httpResponse.addHeader("Content-Disposition", "attachment; filename=\"" + hash + "\"");
+            httpResponse.addHeader("Content-Length", String.valueOf(inputStream.getSize()));
+            inputStream.transferTo(outputStream);
+            outputStream.flush();
+        } catch (IOException e) {
+            throw new SwProcessException(ErrorType.STORAGE, "pull file from storage failed", e);
+        }
+    }
+
+
+    @Operation(summary = "Sign uris to get a batch of temporarily accessible links",
+            description = "Sign uris to get a batch of temporarily accessible links")
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "ok")})
+    @PostMapping(
+            value = "/project/{projectUrl}/evaluation/{version}/uri/sign-links",
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @PreAuthorize("hasAnyRole('OWNER', 'MAINTAINER', 'GUEST')")
+    ResponseEntity<ResponseMessage<Map<String, String>>> signLinks(
```

Comment:
How about move this entrypoint to `FileStorageController`, since it is really the same with dataset link sign. The only benefit for putting it here is that this entrypoint could be authorized. But this authorization has vulnerability that could be passed around. For example, I could pass the project of mine, and the uri of yours.
It doesn't make things worse to move this sign entrypoint to `FileStorageController` since the uri is really hard to guess.